### PR TITLE
chore(docs): polish landing page wording & examples

### PR DESCRIPTION
**Pull request by PaulRBerg**
Feedback for the docs site.

I removed the `It should never revert` branch in order to provide a simpler, gentler example.

### CODE REVIEW

File: docs/src/app/page.tsx [hunk 1]
```diff
@@ -44,7 +44,7 @@ const HeroSection = () => {
       <div className="relative z-10 text-center">
         <h1 className="text-4xl sm:text-5xl md:text-6xl font-bold font-mono mb-4">bulloak</h1>
         <p className="text-lg sm:text-xl mb-8">
-          A test generator based on the Branching Tree Technique
+          A Solidity test generator based on the Branching Tree Technique
```
Comment (alexfertel) [inline docs/src/app/page.tsx:47]: It's no longer only `Solidity`! There's also [poinciana](https://github.com/ericnordelo/poinciana), which generates Cairo tests.

File: docs/src/app/page.tsx [hunk 2]
```diff
@@ -112,26 +111,26 @@ const BulloakFeaturesSection = () => {
...
 ├── Scaffold Command
-│   ├─── Automatically generates test files from .tree specifications
+│   ├─── Generates Solidity test files from .tree specifications
 │   │   ├── Creates modifiers for conditions
 │   │   └── Generates test functions for actions
 │   ├─── Reports syntax errors in your specification
 │   └─── Provides a full AST for easy extension
 ├── Check Command
-│   ├── Ensures code matches its specification
+│   ├── Ensures Solidity implementation matches its specification
```
Comment (alexfertel) [inline docs/src/app/page.tsx:127]: Ditto

**Review by alexfertel — CHANGES_REQUESTED** [review]
Looks good! Thanks for taking the time.

Only thing is that it's no longer only for Solidity!

**Comment by alexfertel** [comment]
Also, I think this is unintentional, right?
Screenshot 2024-08-29 at 20 26 15

**Comment by alexfertel** [comment]
Happy to take over these fixes, lmk.

File: docs/src/app/page.tsx (hunk 1 above)
Comment (PaulRBerg) [reply docs/src/app/page.tsx:47]: Oh wow! I didn't know that.

**Comment by PaulRBerg** [comment]
Resolved now.

**Review by alexfertel — APPROVED** [review]
Awesome! Thanks for taking the time 🚀

**Pull request merged**